### PR TITLE
create-replacement-pr: add provenance

### PR DESCRIPTION
**Pull request by woodruffw**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

This should be the last of the workflows to need a provenance step 🙂 

NB: This one does what `publish-commit-bottles.yml` originally did, i.e. it doesn't have the `pr-pull`/`pr-upload` split. But I can add that in as well, at the cost of a slightly larger diff.

### CODE REVIEW

**Review by carlocab — COMMENTED** [review]
We probably want to split up `pr-pull` and `pr-upload` here like in `publish-commit-bottles.yml`.

**Comment by woodruffw** [comment]
> We probably want to split up `pr-pull` and `pr-upload` here like in `publish-commit-bottles.yml`.

Cool, I can update with that in a bit.

**Review by p-linnane — APPROVED** [review]

**Review by carlocab — APPROVED** [review]

File: .github/workflows/create-replacement-pr.yml [hunk 1]
```diff
@@ -150,29 +152,47 @@ jobs:
         with:
           signing_key: ${{ secrets.BREWTESTBOT_GPG_SIGNING_SUBKEY }}
 
-      - name: Pull PR${{ inputs.upload && ' and upload bottles to GitHub Packages' || '' }}
+      - name: Pull PR
+        id: pr-pull
         working-directory: ${{ steps.set-up-homebrew.outputs.repository-path }}
         env:
-          BREWTESTBOT_NAME_EMAIL: "${{ steps.git-user-config.outputs.name }} <${{ steps.git-user-config.outputs.email }}>"
-          HOMEBREW_GPG_PASSPHRASE: ${{ secrets.BREWTESTBOT_GPG_SIGNING_SUBKEY_PASSPHRASE }}
```
Comment (carlocab) [inline .github/workflows/create-replacement-pr.yml:157]: Thinking about this some more, I suspect we need this (and `--committer`) if `autosquash` is enabled. Thoughts, @Bo98?

Comment (Bo98) [reply .github/workflows/create-replacement-pr.yml:157]: yes

**Pull request merged**